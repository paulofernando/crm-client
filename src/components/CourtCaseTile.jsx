import React from 'react';
import './Style.css';

import ContactTileSmall from "../components/ContactTileSmall";
import NumberFormat from 'react-number-format';

class CourtCaseTile extends React.Component {
    render() {
        return (
            <div className="courtCaseCard">
                <div>                    
                    <div className="courtCaseValue"><NumberFormat value={this.props.courtCase.value} displayType={'text'} thousandSeparator={true} prefix={'$'} /></div>
                    <h3>{this.props.courtCase.title}</h3>                    
                    <p>{this.props.courtCase.description}</p>
                    Contacts:
                    <div className="cardContainer">
                        {this.props.courtCase.contacts &&
                            this.props.courtCase.contacts.map(contact => (                
                                <ContactTileSmall key={contact.id} contact={contact} />
                            ))}
                    </div>
                    <div>{this.props.courtCase.courtDate}</div>
                </div>
            </div>
        );
    }
}

export default CourtCaseTile;