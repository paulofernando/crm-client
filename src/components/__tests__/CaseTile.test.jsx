import React from 'react';
import { shallow, mount } from 'enzyme';

import CourtCaseTile from '../case/CaseTile';

// -------------------------------

const courtCase = {
    title: "Test title",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin arcu enim, tincidunt fringilla rhoncus vitae, semper quis quam. Nam et tincidunt ex. Nunc ut ante id lorem consectetur vulputate ut non nulla. Curabitur viverra, libero vel rutrum volutpat, erat orci eleifend sem, sit amet vestibulum leo lectus at quam",
    value: 1000.00,
    courtDate: "2020-01-01T00:00:00.000Z"
}

const CourtCaseTileComponent = (<CourtCaseTile courtCase={courtCase} />)

// -------------------------------

describe('CourtCaseTile', () => {
    it('renders correctly', () => {
        const container = shallow(CourtCaseTileComponent);        
        expect(container.html()).toMatchSnapshot();
        
        const component = mount(
            <CourtCaseTile courtCase={courtCase} />
        );

        const courtCaseTitle = component.find('.courtCaseTitle');
        expect(courtCaseTitle.text()).toBe('Test title');

        const courtCaseDescription = component.find('.courtCaseDescription');
        expect(courtCaseDescription.text()).toBe('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin arcu enim, tincidunt fringilla rhoncus vitae, semper quis quam. Nam et tincidunt ex. Nunc ut ante id lorem consectetur vulputate ut non nulla. Curabitur viverra, libero vel rutrum volutpat, erat orci eleifend sem, sit amet vestibulum leo lectus at quam');        
    });
});