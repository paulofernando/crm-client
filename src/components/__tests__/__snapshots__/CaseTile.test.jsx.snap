// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourtCaseTile renders correctly 1`] = `"<div class=\\"courtCaseCard\\"><a style=\\"color:white\\" href=\\"/case/undefined\\"><h3 class=\\"courtCaseTitle\\">Test title</h3></a><p class=\\"truncate courtCaseDescription\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin arcu enim, tincidunt fringilla rhoncus vitae, semper quis quam. Nam et tincidunt ex. Nunc ut ante id lorem consectetur vulputate ut non nulla. Curabitur viverra, libero vel rutrum volutpat, erat orci eleifend sem, sit amet vestibulum leo lectus at quam</p><div class=\\"cardContainer\\"></div><time dateTime=\\"1577836800000\\">December 31st 2019</time></div>"`;
